src/main/java/blockchain/Parameters.java

```diff
@@ -11,6 +11,8 @@
 	private int InitialBalance = 20; // balance to start with at launch
 	private int TxMin = 4; // minimum number of Tx in a block
 	private boolean Mode = true; // honest or malicious
+	private int Token = 20; // a token is a representation of any quantity of an asset which the node owns 
+	private boolean Chain = true; // Mode of lightchain (false -> original || true -> Smartcontract)
 	
```

Comment:
Please define final variable like this:
```
public final boolean CONTRACT_MODE = true;
```

Then whenever you use `setChain` refactor it with either `setChain(CONTRACT_MODE)` or `setChain(!CONTRACT_MODE)`, instead of plain true or false. Also, we should do the same when reading from `getChain`. Remember that the developers may not have any idea what would the return value of `getChain` mean, so we should shed some lights on it!.

Reply:
done

Reply:
Please pay closer attention to what I have written: I asked to define a **final** variable as followings: 
```
public final boolean CONTRACT_MODE = true;
```

What does a _final_ variable mean? It means that we cannot change its variable after initialization. So, if you would define `CONTRACT_MODE` as above, it would remain **always true**. 

Now why should we have such a variable? because we want to make an alias name for `true`. But why? because in the context of LightChain, having the `chain` as true or false _does not convey anything to the developer_. What we meant by defining a `chain` variable is to distinguish between the contract mode and non-contact mode. Hence, we should define proper alias that _reflects_ this design decision. When you do so, then instead of having such `if` statements that _does not convey the complete logic_:
```
        if (params.getChain()) {
            cv = new ContractCV(this); 
        }
```
you will be having more meaningful `if` statements:
```
        if (params.getChain() == Parameters.CONTRACT_MODE) {
            
        }
```
⚠️ this is not the only place we should change. We should change _everywhere in the code_ that we treat `setChain` and `getChain` by true or false.

Reply:
We cannot set the value of the contract mode as Final, because we will be passing a new value from the **simulation.config** file.

Reply:
If set final then the compiler throws an error because we are changing the variable's value using setchain() function.

`public void setChain(boolean chain) {
        CONTRACT_MODE = chain;
    }`

Reply:
https://github.com/yhassanzadeh13/lightchain-container/blob/Kedar-smart-contract/simulation.config

Reply:
Check line no. 12 on this link.

Reply:
@KedarKshatriya this comment has been executed wrongly from the beginning. Please read [my first comment on this thread](https://github.com/yhassanzadeh13/lightchain-container/pull/13/files#r588924395). I did not mean to rename the **member field** of this class to `CONTRACT_MODE`. Indeed, we should never use all uppercase naming for a member field in java as a best practice. What I meant was to keep the local member field as it is i.e., 
`private boolean Chain = true; // Mode of lightchain (false -> original || true -> Smartcontract)`

but instead of directly assigning a `true` or `false` variable which does not mean anything for the developer who reads the code, define a `true` value **constant** as:
`public final boolean CONTRACT_MODE = true;`

Then instead of doing `setChain(true)` that again does not convey any meaning to the code, use `setChain(CONTRACT_MODE)` and similarly instead of `setChain(false)` use `setChain(!CONTRACT_MODE)`. So here for example, we do not need to change the body of any of the getters or setters, e.g., `setChain` should be like this:

```
public void setChain(boolean chain) {
    this.Chain = chain;
}
```

But the way we are invoking it should be changed as I explained above, i.e., `setChain(CONTRACT_MODE)`.

What has caused you confusion is that you initially refactored `Chain` to `CONTRACT_MODE` and skipped a major part of my comment which says _please define a final variable_ (not saying _please rename this variable_). 

Thanks.

Reply:
Thank you for clearing out the confusion. It has been fixed here: 5ed56d33d58e6531243fcc678bda7db1f3eb7cf2